src/Controller/Controller.php

```diff
@@ -243,6 +244,11 @@ public function __construct(Request $request = null, Response $response = null,
             $this->name = substr($name, 0, -10);
         }
 
+        if ($componentRegistry !== null) {
+            $this->_components = $componentRegistry;
+            $this->_components->setController($this);
+        }
+
```

Comment:
This is too early. It should be at the same level as the call to `_loadComponents()`, ideally the new variable should be passed to that method.